### PR TITLE
core: bump USB spec version in device descriptor to 2.0

### PR DESCRIPTION
**Pull request by stapelberg**
## Description

The Teensy 4.x comes with USB 2.0 support. This change is required to make the device advertise itself as USB 2.0 in the USB device descriptor.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
n

File: tmk_core/protocol/usb_descriptor.c [hunk 1]
```diff
@@ -351,7 +351,11 @@ const USB_Descriptor_Device_t PROGMEM DeviceDescriptor = {
         .Size                   = sizeof(USB_Descriptor_Device_t),
         .Type                   = DTYPE_Device
     },
+#if defined(USB_VERSION)
+    .USBSpecification           = USB_VERSION,
```
Comment (drashna) [inline tmk_core/protocol/usb_descriptor.c:355]: I'd almost prefer:
```suggestion
#if defined(USB_VERSION_2)
    .USBSpecification           = VERSION_BCD(2, 1, 0),
```

Comment (stapelberg) [reply tmk_core/protocol/usb_descriptor.c]: Done

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Honestly, I'm not exactly happy with the name for the define, but ... I suck at naming. 

Also, could this be documented in: https://github.com/qmk/qmk_firmware/blob/master/docs/config_options.md

**Comment by stapelberg** [comment]
Sure. Updated the docs.

**Review by fauxpark — COMMENTED** [review]
I'm kinda wondering if this even needs a define - all of the MCUs these descriptors currently cover seem to comply with USB 2.0 FS. My previous assumption was this value had to be 1.1 because a USB 2.0 device must also provide a device qualifier descriptor, but I'm now thinking that's only if it is capable of running at high speed *and* full speed:

https://www.keil.com/pack/doc/mw/USB/html/_u_s_b__device__qualifier__descriptor.html
> A high-speed capable device that has different device information for full-speed and high-speed must have a Device Qualifier Descriptor (USB_DEVICE_QUALIFIER_DESCRIPTOR). For example, if the device is currently operating at full-speed, the Device Qualifier returns information about how it would operate at high-speed and vice-versa.

What happens if you define this for, say, an ATmega32U4? If the Windows, macOS and Linux don't complain, I'd say just change it...

**Comment by stapelberg** [comment]
> What happens if you define this for, say, an ATmega32U4

Unfortunately, I don’t have an ATmega32U4 handy. Could you test it, or should we file an issue and address that separately from this change at a later point?

**Comment by fauxpark** [comment]
Just tried on my Satan, it (and/or Windows) didn't like `2, 1, 0` but `2, 0, 0` seems to work fine.
image

**Comment by stapelberg** [comment]
> Some Googling around suggests that 2.10 is not a "real" value, and is what is reported when a USB 3 device is plugged into a USB 2 port. So, that may explain why Windows was confused.

Ah, yeah, https://community.osr.com/discussion/249588/where-is-the-usb-2-1-specification seems to confirm.

Changed it to `2, 0, 0`.

File: tmk_core/protocol/usb_descriptor.c [hunk 2]
```diff
@@ -351,7 +351,11 @@ const USB_Descriptor_Device_t PROGMEM DeviceDescriptor = {
         .Size                   = sizeof(USB_Descriptor_Device_t),
         .Type                   = DTYPE_Device
     },
+#if defined(USB_VERSION_2)
+    .USBSpecification           = VERSION_BCD(2, 0, 0),
+#else
     .USBSpecification           = VERSION_BCD(1, 1, 0),
+#endif
```
Comment (fauxpark) [inline tmk_core/protocol/usb_descriptor.c:358]:
```suggestion
    .USBSpecification           = VERSION_BCD(2, 0, 0),
```

File: docs/config_options.md [hunk 3]
```diff
@@ -105,6 +105,8 @@ This is a C header file that is one of the first things included, and will persi
   * sets the USB polling rate in milliseconds for the keyboard, mouse, and shared (NKRO/media keys) interfaces
 * `#define USB_SUSPEND_WAKEUP_DELAY 200`
   * set the number of milliseconde to pause after sending a wakeup packet
+* `#define USB_VERSION_2`
+  * set the USB device descriptor to USB version 2.
```
Comment (fauxpark) [inline docs/config_options.md:109]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by stapelberg** [comment]
Done

**Review by fauxpark — APPROVED** [review]

**Pull request merged**